<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- compared methods -->
	<node pkg="gyrodometry" name="gyrodometry3d" type="gyrodometry3d">
		<remap from="/odom" to="/odom/neu"/>
		<remap from="/imu/data" to="/imu/data/neu"/>
		<param name="wait_inipose" type="bool" value="false"/>
	</node>

	<!-- msg conversion -->
	<!-- <node pkg="msg_conversion" name="combine_orientation_linear" type="combine_orientation_linear"> -->
	<!-- 	<remap from="/odom" to="/odom/neu"/> -->
	<!-- 	<remap from="/orientation" to="/ekf/quat_rpy"/> -->
	<!-- </node> -->
	<node pkg="msg_conversion" name="odometry_to_rpy" type="odometry_to_rpy">
		<remap from="/odom" to="/odom/neu"/>
		<remap from="/rpy" to="/gt/rpy"/>
	</node>
	<node pkg="msg_conversion" name="quatstamped_to_rpy" type="quatstamped_to_rpy">
		<remap from="/quat" to="/ekf/quat_rpy"/>
		<remap from="/rpy" to="/ekf/rpy"/>
	</node>
	<node pkg="msg_conversion" type="gravity_to_rpy" name="gravity_to_rpy">
		<remap from="/gravity" to="/dnn/g_vector"/>
		<remap from="/rpy" to="/dnn/rpy"/>
	</node>
	<node pkg="msg_conversion" type="odometry_to_rpy" name="gyrodometry_to_rpy">
		<remap from="/odom" to="/gyrodometry"/>
		<remap from="/rpy" to="/gyrodometry/rpy"/>
	</node>

	<!-- validation -->
	<param name="erase_old_data" type="bool" value="True"/>
	<param name="save_csv" type="bool" value="True"/>
	<param name="interval" type="double" value="0.01"/>
	<param name="/num_sub" type="int" value="3"/>
	<param name="/method0" type="str" value="Gyro+DNN(ours)"/>
	<param name="/method1" type="str" value="DNN"/>
	<param name="/method2" type="str" value="Gyro"/>
	<node pkg="msg_printer" type="compare_rp_with_gt.py" name="compare_rp_with_gt" output="screen">
		<remap from="/truth/rpy" to="/gt/rpy"/>
		<remap from="/estimation0/rpy" to="/ekf/rpy"/>
		<remap from="/estimation1/rpy" to="/dnn/rpy"/>
		<remap from="/estimation2/rpy" to="/gyrodometry/rpy"/>
	</node>
	<!-- <node pkg="msg_printer" type="record_float_msg.py" name="record_float_msg" output="screen"> -->
	<!-- 	<remap from="/float" to="/dnn/mul_sigma"/> -->
	<!-- </node> -->
</launch>
